frontend/public/components/environment.jsx

```diff
@@ -85,9 +92,9 @@ export const EnvironmentPage = connect(stateToProps)(
       this.saveChanges = this._saveChanges.bind(this);
       this.updateEnvVars = this._updateEnvVars.bind(this);
 
-      const currentEnvVars = envVarsToArray(this.props.rawEnvData);
+      const currentEnvVars = envVarsToArray(_.cloneDeep(this.props.rawEnvData));
```

Comment:
Can `envVarsToArray` call `_.cloneDeep` so we don't need to repeat it so much?

Reply:
sure thing. all set.